<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>configurator</class>
 <widget class="QMainWindow" name="configurator">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>613</width>
    <height>882</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>configurator</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>OnStep_Logo_Medium.png</normaloff>OnStep_Logo_Medium.png</iconset>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string/>
        </property>
        <property name="pixmap">
         <pixmap>OnStep_Logo_Medium.png</pixmap>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_14">
        <property name="font">
         <font>
          <family>Waree</family>
          <pointsize>16</pointsize>
          <weight>75</weight>
          <italic>false</italic>
          <bold>true</bold>
         </font>
        </property>
        <property name="text">
         <string>Configuration Parameters</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>Mount Type</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="mount_combo">
          <item>
           <property name="text">
            <string>Equatorial</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Alt-Azimuth</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Fork</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>Board Type</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="board_combo">
          <item>
           <property name="text">
            <string>RAMPS 1.4 or 1.5</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MiniPCB (2 axis)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>MaxPCB (4 axis)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>STM32F1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TivaC</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Classic</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Mega2560 Alternate</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Custom</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Max Rate Ratio</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="maxrate_spin">
          <property name="maximum">
           <number>128</number>
          </property>
          <property name="value">
           <number>32</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="QLabel" name="label_9">
          <property name="text">
           <string>PEC Buffer Size</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="pec_spin">
          <property name="maximum">
           <number>100000</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_5">
        <item>
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Auto Sideral Traking mod</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="comboBox_11">
          <item>
           <property name="text">
            <string>Off</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>On</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="2" column="0">
         <widget class="QLabel" name="label_4">
          <property name="text">
           <string>Gears/Pulleys Ratio</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QDoubleSpinBox" name="axis1_worm_spindouble">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>99999.990000000005239</double>
          </property>
          <property name="value">
           <double>144.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QDoubleSpinBox" name="axis1_gear_spindouble">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>99999.990000000005239</double>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QDoubleSpinBox" name="axis2_gear_spindouble">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>99999.990000000005239</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>MicroSteps Tracking</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Axis 1</string>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Axis 2</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QDoubleSpinBox" name="axis2_worm_spindouble">
          <property name="decimals">
           <number>3</number>
          </property>
          <property name="minimum">
           <double>1.000000000000000</double>
          </property>
          <property name="maximum">
           <double>99999.990000000005239</double>
          </property>
          <property name="value">
           <double>144.000000000000000</double>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="axis1_motor_combo">
          <item>
           <property name="text">
            <string>200 steps, 1.8°/step</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>400 steps, 1.8°/step</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>48 steps, 7.5°/step</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="1">
         <widget class="QComboBox" name="axis1_driver_combo">
          <item>
           <property name="text">
            <string>A4988</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DRV8825</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>LV8729 or RAPS128</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2208</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (Quiet)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (VQuiet)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (Quiet, LowPWR)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (VQuiet, LowPWR)</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>Stepper-Steps</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QComboBox" name="axis2_motor_combo">
          <item>
           <property name="text">
            <string>200 steps, 1.8°/step</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>400 steps, 0.9°/step</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>48 steps, 7.5°/step</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="6" column="2">
         <widget class="QComboBox" name="axis2_driver_combo">
          <item>
           <property name="text">
            <string>A4988</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>DRV8825</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>LV8729 or RAPS128</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2208</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (Quiet)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (VQuiet)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (Quiet, LowPWR)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>TMC2130 (VQuiet, LowPWR)</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Worm Wheel Ratio</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="axis1_micro_combo">
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>32</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>64</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>128</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>256</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="2">
         <widget class="QComboBox" name="axis2_micro_combo">
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>32</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>64</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>128</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>256</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>MicroSteps Slewing</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>Driver Model</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QComboBox" name="axis1_mod_combo">
          <item>
           <property name="text">
            <string>Same as tracking</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Off</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="5" column="2">
         <widget class="QComboBox" name="axis2_mod_combo">
          <item>
           <property name="text">
            <string>Same as tracking</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Off</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QPushButton" name="generate_buttom">
        <property name="text">
         <string>Generate</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout_6">
        <item>
         <widget class="QLineEdit" name="path_line"/>
        </item>
        <item>
         <widget class="QToolButton" name="path_button">
          <property name="text">
           <string>...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextBrowser" name="prompt_textbrowser">
        <property name="html">
         <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Configure your setup for Onstep !&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Standby...&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QDialogButtonBox" name="buttonBox">
      <property name="standardButtons">
       <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>613</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuOnStep_Configurator">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionLoad"/>
   </widget>
   <widget class="QMenu" name="menuInfo">
    <property name="title">
     <string>Info</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuOnStep_Configurator"/>
   <addaction name="menuInfo"/>
  </widget>
  <action name="actionSave">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionLoad">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Load</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>clicked(QAbstractButton*)</signal>
   <receiver>configurator</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>556</x>
     <y>844</y>
    </hint>
    <hint type="destinationlabel">
     <x>555</x>
     <y>858</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
